infra-as-code/bicep/ai-agent-service-dependencies.bicep

```diff
@@ -24,6 +24,16 @@ param privateEndpointSubnetResourceId string
 
 // ---- New resources ----
 
+@description('Deploy Azure AI Search instance for the Azure AI Foundry Agent Service (dependency). This is used when a user uploads a file to the agent, and the agent needs to search for information in that file.')
+module deployUserManagedIdentity 'ai-agent-user-managed-identity.bicep' = {
+  name: 'agentUserManagedIdentityDeploy'
+  scope: resourceGroup()
+  params: {
+    location: location
+    baseName: baseName
+  }
+}
```

Comment:
I still don't understand why this is a submodule.  Why not bring the identity creation directly into this file?  What's the benefit of having an extra file for this vs the tradeoff of it?